server/src/main/java/org/opensearch/indices/replication/checkpoint/ReplicationCheckpoint.java

```diff
@@ -30,37 +32,61 @@ public class ReplicationCheckpoint implements Writeable, Comparable<ReplicationC
     private final long segmentsGen;
     private final long segmentInfosVersion;
     private final long length;
+    private final String latestSupportedCodec;
+    private CodecService codecService;
 
     public static ReplicationCheckpoint empty(ShardId shardId) {
         return new ReplicationCheckpoint(shardId);
     }
 
     private ReplicationCheckpoint(ShardId shardId) {
+        codecService = new CodecService(null, null);
```

Comment:
The problem here is that plugins can implement a custom CodecService provided through [`getCustomCodecServiceFactory`](https://github.com/opensearch-project/OpenSearch/blob/8c9cfdcf77d4a13f74fb010c235ab307b9fe6cc7/server/src/main/java/org/opensearch/plugins/EnginePlugin.java#L87). When you create a new ReplicationCheckpoint I think you'll need to use the codecService defined by the indexSettings?

Reply:
Modified the constructor for ReplicationCheckpoint to pick up the codecService set up for the shard.

Reply:
Thanks for the review @nknize! Could you take a second look to see if the changes address your concerns?